### ISSUE TITLE
Bookworm?

### ISSUE THREAD

**Issue by martin-braun** [pimox/pimox7#145]
Is support for Debian 12 (Bookworm) planned? Do we have to wait for Proxmox to officially ship with Bookworm? There seem to be [official instructions](https://pve.proxmox.com/wiki/Install_Proxmox_VE_on_Debian_12_Bookworm) on getting Proxmox to run on Bookworm.

Thanks.

**Comment by TheBossME** [comment]
Read my comments, there is a stable proxmox 8.0-3 for arm64

Search for arm64 unofficial proxmox

Good luck.

And use this 
https://raspi.debian.net/tested/20230612_raspi_4_bookworm.img.xz

With this repo 

https://github.com/jiangcuo/Proxmox-Arm64

**Comment by martin-braun** [comment]
Very helpful @TheBossME, dankeschön. :)

**Comment by TheBossME** [comment]
Hi Martin,

I will create a YT Video with all the needed steps to prepare a stable Proxmox installation while adding

deb https://mirrors.apqa.cn/proxmox/ pvearm8 main

Repository. I'm not using the pvetest repo. 

The video will be in German language and maybe with English comments too. 

If you have suggestions or questions, feel free to contact me.

Regards Beatrice

**Comment by AubsUK** [comment]
Debian ISO (what I use):
https://cdimage.debian.org/debian-cd/current/arm64/iso-cd/

This is a good video on installing the latest Ubuntu, the exact same principle can be used for Debian.
https://youtu.be/A4OhdsYULfk

A very brief summary and quick steps to get it working

Download the ISO
New VM
Set ID/Name
OS: Do not use media
BIOS: UEFI
Disk 10GB
CPU 2
CPU Type: Host

Hardware >> Remove DC/DVD
Add >> CD/DVD Drive >> Set to SCSI >> Choose ISO
Add >> Serial Port  >> 0 (for initial setup)
Display >> Graphic card >> Serial terminal 0
Options >> Boot order >> CD to the top
Start
Console >> xterm
Install

Close terminal, shutdown VM
Hardware >> CD >> Do not use media
Double click Display >> Default
Serial Port >> Remove

**Comment by martin-braun** [comment]
@TheBossME Thanks a lot Beatrice.

> If you have suggestions or questions, feel free to contact me.

What is the name of your YouTube channel? I'd like to subscribe. I'm excited to see your video guide. :)

**Comment by cvocvo** [comment]
> Read my comments, there is a stable proxmox 8.0-3 for arm64
> 
> Search for arm64 unofficial proxmox
> 
> Good luck.
> 
> And use this https://raspi.debian.net/tested/20230612_raspi_4_bookworm.img.xz
> 
> With this repo
> 
> https://github.com/jiangcuo/Proxmox-Arm64

Is it possible to upgrade to this via an existing pimox7 install? Or do we have to do a clean install?

**Comment by DonnieDice** [comment]
I had to manually install "libssl1.1_1.1.0g-2ubuntu4_arm64.deb" and "libtpms0_0.9.2-3.1_arm64.deb" to get the install to work.
wget
https://ubuntu.pkgs.org/18.04/ubuntu-main-arm64/libssl1.1_1.1.0g-2ubuntu4_arm64.deb.html
https://debian.pkgs.org/12/debian-main-arm64/libtpms0_0.9.2-3.1_arm64.deb.html
dpkg
this stuff.

**Comment by cvocvo** [comment]
> I had to manually install "libssl1.1_1.1.0g-2ubuntu4_arm64.deb" and "libtpms0_0.9.2-3.1_arm64.deb" to get the install to work. wget https://ubuntu.pkgs.org/18.04/ubuntu-main-arm64/libssl1.1_1.1.0g-2ubuntu4_arm64.deb.html https://debian.pkgs.org/12/debian-main-arm64/libtpms0_0.9.2-3.1_arm64.deb.html dpkg this stuff.

To clarify did you do something like this? (Just trying to figure out the right sequence of steps to upgrade/migrate)
 - Have an existing pimox7 install
 - Then upgrade debian from 11 to 12 (and install those packages you linked)
 - Then change the Proxmox repos from the Pimox7 ones to the ones listed here: https://github.com/jiangcuo/Proxmox-Arm64/wiki/Install-Proxmox-VE-on-Debian-bullseye#install-proxmox-ve
 - Then updated packages (thus upgrading proxmox to v8)

**Comment by DonnieDice** [comment]
Ok ignore that, I was doing something wrong. I ended up getting things running using 2023-05-03-raspios-bullseye-arm64-lite.img.xz and Virtual Environment 7.4-16.

**Comment by prafiles** [comment]
Thanks guys! I got it working using the above mentioned links with slight modification. I've posted details on my blog [here](https://prafiles.in/proxmox8-on-arm64/).

Basically in place upgrade does work, however I needed the apt mirror to be:
`deb https://mirrors.apqa.cn/proxmox/debian/pve bookworm port`

After the upgrade, I'd to run `apt-get autoremove` which fixed some broken dependencies of things not getting compiled by the kernel automatically.

**Comment by Davidorado1** [comment]
I got it working too. I used mainly [this](https://gist.github.com/gtx28/059ac9a0c85e722225c6d5150d659dec) manual to do it.
But I changed the order a bit:
In one of my two raspi/Proxmox Setups I got an error while upgrading from bullseye to bookworm that I apparently wanted to uninstall proxmox. I stopped at this point and first changed the proxmox repository from the pimox7 to the other one:
deb https://global.mirrors.apqa.cn/proxmox/debian/pve bookworm port
After the next dist-upgrade, apt autoremove, reboot and another apt upgrade everything went really well.

**Comment by martin-braun** [comment]
@prafiles @Davidorado1 Not trying to imply something personal about you guys, but you both seem to emphasize that only the Chinese mirror is working. I'm happy that you report your success story, however:

It's a mirror, it shouldn't make a difference. This looks suspicious to me. I wouldn't want to take a mirror from China, not because of their people, but because of their rulers.

**Comment by Davidorado1** [comment]
Well, I understand your concerns, but why do you come across this issue 2 months later? 
@TheBossME posted [this](https://github.com/jiangcuo/Proxmox-Arm64) github repository in his first answer. Did you even take a look at the repo and the wikipage? The guy behind the repository is the one who provides this mirror (as you can read [here](https://github.com/jiangcuo/Proxmox-Arm64/issues/37) ).
Sure, it can be suspicious, but that shouldn´t be new to you.
Have a nice weekend!

**Comment by martin-braun** [comment]
@Davidorado1 Sorry, you are right. Tbh, I wasn't really concerned about that 2 months earlier until I gained some more knowledge about the current situation over there. But yes, the second link explains the mirror situation, which isn't a mirror by this definition. I didn't look into its details back then as I'm still busy on other stuff.

Thanks for sharing this with me though. Hopefully jiangcuo is legit. Wish you a nice weekend as well.

**Comment by prafiles** [comment]
> @prafiles @Davidorado1 Not trying to imply something personal about you guys, but you both seem to emphasize that only the Chinese mirror is working. I'm happy that you report your success story, however:
> 
> It's a mirror, it shouldn't make a difference. This looks suspicious to me. I wouldn't want to take a mirror from China, not because of their people, but because of their rulers.

I agree with you 100%. I also do not wish to encourage the chinese mirror. I am indeed quite uncomfortable with the same.

However that is the only one that worked for me including trying CDN DNS Servers like `1.1.1.1` as well as my usual self-hosted recursive Unbound server. I tried multiple things, but as discussed above, I was not comfortable with the path of removing all of my packages and then install pve server. If an alternate path works, I'll readily retry and will be glad to assist in anyway I can.

My intent to share this result was to maybe help point someone in the direction to help them to do so. If it helps, I am based out of India, if mirrors behave differently for different locations.

**Comment by martin-braun** [comment]
@prafiles Thank you for the feedback, which confirms my conclusion. Although called "mirror", it's a custom repository adjusted by jiangcuo to work with the PI. I investigated his Chinese blog, he seems to be legit, however we know how relationships between companies and governments are quite different in China compared to the Western world, we wouldn't even noticed interference by higher ups, unless you have the time and dedication to audit all the package sources in that "mirror".

I will keep this open for the sake of finding "official" alternatives (as official as it can get).
